Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.lean

```diff
@@ -625,7 +625,7 @@ variable {n : MeasurableSpace β} in
 lemma Finset.stronglyMeasurable_prod_apply {ι : Type*} {f : ι → α → β → M} {g : α → β}
     {s : Finset ι} (hf : ∀ i ∈ s, StronglyMeasurable ↿(f i)) (hg : Measurable g) :
     StronglyMeasurable fun a ↦ (∏ i ∈ s, f i a) (g a) := by
-  simp; fun_prop (discharger := assumption)
+  simp only [Finset.prod_apply]; fun_prop (discharger := assumption)
```

Comment:
Just out of curiosity, doesn't fun_prop already look at assumptions?

Reply:
Nope, it doesn't: removing the custom discharger makes the proof fail.

Reply:
I wonder if it should do that, it seems like a conservative yet helpful modification